panel/backends/wayland/kwin_wayland/lxqtwmbackend_kwinwayland.cpp

```diff
@@ -123,6 +123,15 @@ bool LXQtWMBackend_KWinWayland::supportsAction(WId windowId, LXQtTaskBarBackendA
         state = LXQtTaskBarPlasmaWindow::state::state_fullscreenable;
         break;
 
+    case LXQtTaskBarBackendAction::DesktopSwitch:
```

Comment:
Hi Marcus. I think we should split "Switch Desktop" functionality used by Desktop Switch plugin (and called with null `windowId`) from "Move window to desktop" functionality used by Taskbar plugin.
The first one is handled in the if case above before checking if `windowId` is valid otherwise it would always return false.

Reply:
I am sorry. I did not quite get what you mean by

> The first one is handled in the if case above before checking if windowId is valid otherwise it would always return false.

Do you mean to say that when `windowId` is not nullptr (i.e., when we're trying to move a window to desktop), it should return false?

Reply:
> Do you mean to say that when `windowId` is not nullptr (i.e., when we're trying to move a window to desktop), it should return false?

What I meant is that switching desktops and moving windows on some desktop are 2 different functionalities which might get implemented with different protocols.
So it would be better to distinguish them and have 2 different enum values.
Currently `LXQtTaskBarBackendAction::DesktopSwitch` is not used at all by taskbar so `supportsAction()` will not be queried before moving windows to another desktop.
It is only used by the DesktopSwitch plugin.

Reply:
I'll open a separate PR for this, and include this check. I did not think of it because it wasn't necessary - wlroots does not support virtual desktops so, you'll always get 1 desktop, anyways.